tests/rptest/services/databricks_workspace.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2025 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import uuid
+from dataclasses import dataclass
+from typing import Set
+
+import databricks
+import databricks.sdk
+import databricks.sdk.errors
+import databricks.sql
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.catalog import (
+    CatalogInfo,
+    CatalogType,
+    ExternalLocationInfo,
+)
+from ducktape.services.service import Service
+from ducktape.tests.test import TestContext
+
+from rptest.context.databricks import DatabricksContext
+
+
+class DatabricksWorkspace(Service):
+    """
+    A service wrapper for a hosted Databricks workspace.
+    This service does not manage any nodes, but it provides a wrapper for
+    the Databricks SDK to interact with the workspace and manage its resources.
+    Including post-testing cleanup.
+    """
+    def __init__(self, context: TestContext):
+        super().__init__(context, num_nodes=0)
+
+        self._databricks_context = DatabricksContext.from_context(context)
+
+        self._client = WorkspaceClient(
+            host=self._databricks_context.server_hostname,
+            token=self._databricks_context.token,
+        )
+
+        # Fail fast if credentials are invalid.
+        self._verify_credentials()
+
+        # Resource trackers.
+        self._location_names: Set[str] = set()
+        self._catalog_names: Set[str] = set()
+
+    def stop(self, **kwargs):
+        self.logger.debug("Stopping Databricks workspace")
+        self._cleanup_catalogs()
+        self._cleanup_dbx_locations()
+        super().stop(**kwargs)
+
+    def create_catalog(self, bucket: str) -> "DatabricksCatalogInfo":
```

Comment:
Still orienting myself a bit, is the expectation that each bucket will be used by exactly one catalog? And e.g. we don't need to set the location to some s3://bucket/catalog_base_location?

Reply:
I believe so, yes. We have one bucket per test. Databricks seems to supports sub-paths too but it doesn't seem necessary for now.